pyiron/dft/job/generic.py

```diff
@@ -116,15 +116,15 @@ def get_structure(self, iteration_step=-1, wrap_atoms=True):
 
         Returns:
             pyiron.atomistics.structure.atoms.Atoms: The required structure
-
-
         """
         snapshot = super(GenericDFTJob, self).get_structure(
             iteration_step=iteration_step, wrap_atoms=wrap_atoms
         )
-        spins = self.get("output/generic/dft/atom_spins")
-        if spins is not None:
-            snapshot.set_initial_magnetic_moments(spins[iteration_step])
+        try:
+            if len(self.get_magnetic_moments()) > 0:
```

Comment:
```
magmoms = self.get_magnetic_moments(iteration_step=iteration_step)
if magmoms is not None: 
    snapshot.set_initial_magnetic_moments(magmoms)
```

Reply:
is already done 😄 Look at the latest commit.

Reply:
I am not a fan of `try except` is there another way how we can handle this?

Reply:
What I would suggest is to implement `get_magnetic_moments` in `dft/job/generic.py` as follows:
```
def get_magnetic_moments(iteration_step=-1):
      spins = self.get("output/generic/dft/atom_spins")
      if spins is not None:
          return spins[iteration_step]
      else:
          return None
```
These changes should work for Sphinx, shouldn't they? Than we do not need the  `try except`.

Reply:
I like this suggestion because it is in agreement with https://pyiron.github.io/source/faq.html#which-output-quantities-are-stored-in-pyiron

Reply:
> I like this suggestion because it is in agreement with https://pyiron.github.io/source/faq.html#which-output-quantities-are-stored-in-pyiron
BTW: There the quantity is given as `atomS_spins`, not `atom_spins`.